apps/cowswap-frontend/src/modules/wallet/web3-react/connection/asyncConnector.ts

```diff
@@ -1,27 +1,50 @@
+import { Web3Provider } from '@ethersproject/providers'
 import { Actions, Connector } from '@web3-react/types'
 
+import EventEmitter from 'events'
+
+export const ASYNC_CUSTOM_PROVIDER_EVENT = 'customProvider'
+
+function initCustomProvider(self: AsyncConnector, connector: Connector, chainId: number) {
+  if (connector.provider && !connector.customProvider) {
+    self.customProvider = new Web3Provider(connector.provider, chainId)
```

Comment:
Just inherited it from:
https://github.com/Uniswap/web3-react/blob/main/packages/core/src/hooks.ts#L366